File: Cpp/Partition-Array-by-Odd-and-Even.cpp
#include <vector>

class Solution {
public:
    /**
     * @param nums: a vector of integers
     * @return: nothing
     */
    void partitionArray(vector<int> &nums) {
        // write your code here
        if (nums.empty()) return;
        int left = 0, right = nums.size() - 1;
        while (left < right) {
            while (nums[left] % 2 == 1) left ++;
            while (nums[right] % 2 == 0) right --;
            if (left < right) {
                int temp = nums[left];
                nums[left] = nums[right];
                nums[right] = temp;
                left ++;
                right --;
            }
        }
        return;
    }
};

File: Python/Partition-Array-by-Odd-and-Even.py
class Solution:
    # @param nums: a list of integers
    # @return: nothing
    def partitionArray(self, nums):
        # write your code here
        left = 0
        right = len(nums) - 1
        while left < right:
            while left < right and nums[left] & 1 == 1:
                left += 1
            while left < right and nums[right] & 1 == 0:
                right -= 1
            if left < right:
                temp = nums[left]
                nums[left] = nums[right]
                nums[right] = temp
